modules/dists/BlockDist.chpl

```diff
@@ -1625,6 +1625,7 @@ proc BlockArr.doiOptimizedSwap(other: this.type)
 // type, stridability, or rank mismatch in the other argument). When
 // debugOptimizedSwap is off, this overload will be ignored due to its where
 // clause.
+pragma "last resort"
 proc BlockArr.doiOptimizedSwap(other) where debugOptimizedSwap {
```

Comment:
So the issue is that if `debugOptimizedSwap` is set this routine is now more preferred over the one taking `other: this.type` in the case where that requires coercions? Would another approach be to rewrite that code as `BlockArr.doiOptimizedSwap(other) where other.type == this.type`? I'm not sure I actually prefer that, just trying to understand the rules.

Reply:
Yes, but if it was relying on implicit conversions before, that would give different behavior.